tools/lsp/language.rs

```diff
@@ -565,6 +603,25 @@ pub(crate) async fn reload_document_impl(
     lsp_diags
 }
 
+pub async fn open_document(
+    ctx: &Rc<Context>,
+    content: String,
+    url: lsp_types::Url,
+    version: Option<i32>,
+    document_cache: &mut DocumentCache,
+) -> common::Result<()> {
+    i_slint_core::debug_log!("OPEN FILE: {url}");
+    ctx.open_urls.borrow_mut().insert(url.clone());
+
+    reload_document(ctx, content, url, version, document_cache).await
+}
+
+pub async fn close_document(ctx: &Rc<Context>, url: lsp_types::Url) -> common::Result<()> {
+    i_slint_core::debug_log!("CLOSE FILE: {url}");
+    ctx.open_urls.borrow_mut().remove(&url);
+    invalidate_document(ctx, url).await
```

Comment:
currently, invalidate_document reload that document. Do we want that?

Reply:
No, that is pretty useless. Will change in the next update.